toolkit/hooks/src/use-date-helper.tsx

```diff
@@ -1,47 +1,82 @@
-import React from "react";
 import { DateTime } from "ts-luxon";
 
-const convertGMTDateToTimezone: React.FC<any> = (
-	date: string | Date,
-	format = "MMM dd, yyyy, hh:mm a",
-) => {
-	if (typeof date === "string") {
-		const dateTime = DateTime.fromISO(date, { zone: "local" }).toFormat(format);
-		return dateTime;
-	} else {
-		const dateTime = DateTime.fromJSDate(date, { zone: "local" }).toFormat(
-			format,
-		);
-
-		console.log(dateTime, "DateTime____");
+function useDateHelpers() {
+	const addLeadingZeroes = (date: string) => {
+		return date.replace(/(\b\d\b)/g, "0$&");
+	};
 
-		return dateTime;
-	}
-};
+	const parseCustomDate = (date: string, format = "dd/MM/yyyy") => {
+		date = addLeadingZeroes(date);
+		const formatParts = format.split(/[- /]/);
+		const dateParts = date.split(/[- /]/);
 
-function useDateHelpers() {
-	const getFormattedDate = (
-		date: string,
-		timestamp = false,
-		format = "MMM dd, yyyy, hh:mm a",
-	) => {
-		console.log("DATE_VALUE", date);
-		if (timestamp) {
-			date = new Date(parseInt(date) * 1000).toISOString();
+		if (formatParts.length !== dateParts.length) {
+			throw new Error("Date format and input date do not match.");
 		}
+		const dateValues: {
+			year: number;
+			month: number;
+			day: number;
+		} = {
+			year: 0,
+			month: 0,
+			day: 0,
+		};
+
+		for (let i = 0; i < formatParts.length; i++) {
+			const formatPart = formatParts[i];
+			const datePart = dateParts[i];
 
-		return convertGMTDateToTimezone(date, format);
+			if (formatPart == "yyyy" || formatPart == "YYYY") {
+				dateValues.year = parseInt(datePart);
+			} else if (
+				formatPart == "MM" ||
```

Comment:
Instead of multiple conditions like this, you can create an array and use includes check here.
```
formatPart == "MM" ||
formatPart == "M" ||
formatPart == "mm" ||
formatPart == "m"
```
MonthFormat = ["MM","M", "mm","m"];
and then use the check 

`MonthFormat.includes(formatPart)`.
This will make it move easily extendable and clean.